code/Weapons/Gadget/Components/AnvilPhysicsGadgetComponent.cs

```diff
@@ -0,0 +1,92 @@
+﻿namespace Grubs;
+
+[Prefab]
+public partial class AnvilPhysicsGadgetComponent : GadgetComponent
+{
+	[Prefab, Net]
+	public bool LockXAxis { get; set; } = true;
+
+	[Prefab, Net]
+	public bool AffectedByWind { get; set; } = false;
+
+	[Prefab, ResourceType( "sound" )]
+	public string CollisionSound { get; set; }
+
+	[Prefab, Net, Predicted]
+	public int Bounces { get; set; } = 0;
+
+	[Prefab, Net]
+	public float BounceForce { get; set; } = 100.0f;
+
+	private bool _isGrounded;
+	private bool _wasGrounded;
+
+	public override bool IsResolved()
+	{
+		return Bounces == 0 && Gadget.Velocity.IsNearlyZero( 2.5f );
+	}
+
+	public override void OnUse( Weapon weapon, int charge )
+	{
+		if ( GrubsConfig.WorldTerrainType is GrubsConfig.TerrainType.Texture )
+			Gadget.Position = Grub.Player.MousePosition.WithZ( GrubsGame.Instance.Terrain.WorldTextureHeight + 192f );
+		else
+			Gadget.Position = Grub.Player.MousePosition.WithZ( GrubsConfig.TerrainHeight + 192f );
```

Comment:
Maybe we should throw this into a method as I imagine getting the correct terrain height will be needed for the future.

GetTerrainHeight(int offset)?

@trundlr is this correct?

Reply:
Currently there is WorldTextureHeight (which needs to be renamed) which gets the height of the level. For texture levels, it is read from the png. For generated levels, it is set to the max value in the heightmap for the z direction. So basically, you can use WorldTextureHeight for both cases here. Knowing this, a function that returns the height with an offset would simply be a wrapper for for WorldTextureHeight + offset, which is not needed.

Reply:
It looks like WorldTextureHeight was just swapped to be the terrain's "maxY" (seems to always return 26) if the terrain is generated, in order to make the menu's Terrain Preview center nicely on the terrain. Should I swap this back and we find another way to center the terrain preview?